featuretools/tests/synthesis/test_dfs_method.py

```diff
@@ -339,6 +339,29 @@ def test_no_warns_with_different_case_string_primitives(es):
         assert not record
 
 
+def test_no_warns_with_and_without_snakecase_string_primitives(es):
+    # Should not raise a UnusedPrimitiveWarning warnings
+    for trans_primitive in ["is_null", "isNull", "IsNull", "_iSnu__Ll_", "i_s_n_u_L_l"]:
```

Comment:
Am I understanding this correctly that we will accept `i_s_n_u_l_l` as a valid primitive name with this change? 

If so, is that what we intend here? That seems a bit too flexible to me.

Reply:
If we don't want to break the current behavior, I think we need to allow `is_null`, but beyond that, it seems like the only other things we should match on is the exact value returned by `primitive.__name__` (`IsNull`). I'm not even sure we should match on camel case, but that just my two cents here.